CMakeLists.txt

```diff
@@ -15,36 +17,65 @@ execute_process(
   ERROR_QUIET)
 
 if(ECODE)
-  message(
-    "Failed to determine version from git, using ${CMAKE_PROJECT_VERSION}")
+  message("Failed to determine version from git, using ${CMAKE_PROJECT_VERSION}")
   set(NEURONPARQUET_VERSION ${CMAKE_PROJECT_VERSION})
 endif()
 
+# Find required packages
 find_package(MPI REQUIRED)
 find_package(Arrow REQUIRED)
+
+# Add MPI include directories and definitions
+include_directories(SYSTEM ${MPI_INCLUDE_PATH})
+add_definitions(${MPI_CXX_COMPILE_FLAGS})
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${MPI_CXX_LINK_FLAGS}")
+add_definitions(-DOMPI_SKIP_MPICXX)
```

Comment:
```suggestion
```
Please link against the CMake targets for MPI. For example like this: https://github.com/BlueBrain/HighFive/blob/9d4c77e719dc3fcad77ba13d809ba2cdf9f845d2/cmake/HighFiveConfig.cmake#L15